clippy_lints/src/enum_sex.rs

```diff
@@ -0,0 +1,57 @@
+use rustc::lint::*;
+use rustc::hir::*;
+use crate::utils::span_lint;
+
+/// **What it does:** Checks for enums named `Sex` or `Gender` with only two variants
+///
+/// **Why is this bad?** This is likely to indicate code that cannot properly
+/// model the data it is trying to represent.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// enum Sex {
+///     Female,
+///     Male,
+/// }
+/// struct Person {
+///     sex: Sex,
+///     // other fields
+/// }
+/// ```
+///
+/// Could be written:
+///
+/// ```rust
+/// struct Person {
+///     sex: &str,
+///     // other fields
+/// }
+/// ```
+declare_clippy_lint! {
+    pub ENUM_SEX,
+    pedantic,
+    "enum named `Sex` or `Gender` with only two variants"
+}
+
+#[derive(Copy, Clone)]
+pub struct EnumSex;
+
+impl LintPass for EnumSex {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(ENUM_SEX)
+    }
+}
+
+impl<'a, 'tcx> LateLintPass<'a, 'tcx> for EnumSex {
+    fn check_item(&mut self, cx: &LateContext, item: &Item) {
+        if item.name.as_str() == "Sex" || item.name.as_str() == "Gender" {
```

Comment:
We should probably create a whitelist of things to lint on, similar to the lint triggering on `foo` and `bar` for bad identifiers (because that lint also triggers on french placeholders). So making this open to multi-language support from the start feels right.

Reply:
Multi-language support is a very good point. I'll see what I can learn from the blacklisted_name lint and see if I can handle it in the same way.